src/App.js

```diff
@@ -1,19 +1,120 @@
 import React from 'react';
 import './App.css';
+import { TodosList } from './components/TodosList/TodosList';
+import { UsersList } from './components/UsersList/UsersList';
 
-import users from './api/users';
+import usersFromApi from './api/users';
+import todosFromApi from './api/todos';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
+const preparedTodos = todosFromApi.map(todo => ({
+  ...todo,
+  user: usersFromApi.find(user => user.id === todo.userId),
+}));
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    selectedUserId: 0,
+    title: '',
+    errorTitle: '',
+    errorSelectUser: '',
+  }
+
+  handleSelectUser = (event) => {
+    const {
+      name, value,
+    } = event.target;
+
+    this.setState({
+      [name]: +value,
+      errorSelectUser: '',
+    });
+  }
+
+  handleChangeTitle = (event) => {
+    const {
+      name, value,
+    } = event.target;
+
+    this.setState({
+      [name]: value,
+      errorTitle: '',
+    });
+  }
+
+  render() {
+    const {
+      todos,
+      title,
+      selectedUserId,
+      errorSelectUser,
+      errorTitle,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <p>
+          <span>Todos: </span>
+          {todos.length}
+        </p>
+
+        <form onSubmit={((event) => {
+          event.preventDefault();
+
+          if (!selectedUserId) {
+            this.setState({
+              errorSelectUser: 'Please choose a user',
+            });
+          }
+
+          if (!title) {
+            this.setState({
+              errorTitle: 'Please enter the title',
+            });
+          }
+
+          if (selectedUserId && title) {
+            this.setState((prevState) => {
+              const newGood = {
+                userId: prevState.selectedUserId,
+                id: todos.length + 1,
+                title: prevState.title,
+                completed: false,
+                user: usersFromApi[prevState.selectedUserId - 1],
```

Comment:
We don't need the whole 'user' object. Just its name

Reply:
Small update:
Actually, we don't need to pass the 'name' only 'userId'.
Each TODO in our list looks like this:
```
 {
    userId: 1,
    id: 1,
    title: 'delectus aut autem',
    completed: false,
  },
```
So we should pass 'userId' to our new TODO-object, as task description says:

> each TODO should have a title and userId selected from a list of given users

Also, you may pass the 'Name' of the user to the TodoList to make it looks prettier, but not to add it to the new 'todo' object and it's not essential for this task